CONTRIBUTING.md

```diff
@@ -2,3 +2,104 @@ Contributing
+
+	class IncrementalDecoder(codecs.IncrementalDecoder):
+    		def decode(self, input, final=False):
+        		return codecs.charmap_decode(input,'ignore',decoding_table)[0]
+
+This should allow the CSS file to be loaded, but may result in icons being rendered improperly. A proper fix will likely involve confirming and correcting the encoding used by the CSS file for the RTD theme.
+
+If contributing new documentation content, please review:
+
+- the [Sphinx Style Guide](http://documentation-style-guide-sphinx.readthedocs.org/en/latest/style-guide.html)
+- [ASP.NET Docs Style Guide](http://docs.asp.net/contribute/style-guide/style-guide)
+
+## Adding Content ##
+
+Before adding content, submit an issue with a suggestion for your proposed article. Provide detail on what the article would discuss, and how it would relate to existing documentation.
+
+Articles should be organized into logical groups or sections. Each section should be given a named folder (e.g. /yourfirst). Within each section, each article should also have its own folder, within which the article.rst file resides. Images and other static resources (code samples, etc.) should be placed in a _static folder within the article folder.
+
+### Example Structure ###
+
+	docs
+		/yourfirst
+			/fundamentalconcepts
+				/_static
+					about-page.png
+					add-config.png
+					...
+				fundamentalconcepts.rst
+			/yourfirstaspnetapplication
+				/_static
+					about-page.png
+					add-api-controller.png
+					...
+				yourfirstaspnetapplication.rst
+
+**Note:** Sphinx will automatically fix duplicate image names, such as the about-page.png files shown above. There is no need to try to ensure uniqueness of static files beyond an individual article.
+
+Author information should be placed in the _authors folder following the example of steve-smith.rst. Place photos in the photos folder - size them to be no more than 125px wide or tall.
+
+## Process for Contributing ##
+
+**Step 1:** Open an Issue describing the article you wish to write and how it relates to existing content. Get approval to write your article.
+
+**Step 2:** Fork the `/aspnet/docs` repo.
+
+**Step 3:** Create a `branch` for your article.
+
+**Step 4:** Write your article, placing the article in its own folder and any needed images in a _static folder located in the same folder as the article. Be sure to follow the [ASP.NET Docs Style Guide](http://docs.asp.net/contribute/style-guide/style-guide). If you have code samples, place them in a folder within the `/samples/` folder.
+
+**Step 5:** Submit a Pull Request from your branch to `aspnet/docs/master`.
+
+**Step 6:** Discuss the Pull Request with the ASP.NET team; make any requested updates to your branch. When they are ready to accept the PR, they will add a :shipit: comment.
```

Comment:
Might also want to put `(:shipit:)` in backticks so that people see what to type in. (You can hover over GitHub emoji to see the text, but I'm guessing a lot of people don't know that.)